types/tx/types.go

```diff
@@ -1,16 +1,106 @@
 package tx
 
 import (
+	fmt "fmt"
+
 	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+// MaxGasWanted defines the max gas allowed.
+const MaxGasWanted = uint64((1 << 63) - 1)
+
 var _, _ codectypes.UnpackInterfacesMessage = &Tx{}, &TxBody{}
+var _ sdk.Tx = &Tx{}
```

Comment:
We can remove [WrapTx](https://github.com/cosmos/cosmos-sdk/blob/9e85e81e0e8140067dd893421290c191529c148c/x/auth/tx/builder.go#L365) now that tx.Tx implements sdk.Tx.

However, in this PR, I just modified the minimal code on Tx to make it work. For clarity, I propose to leave it to another PR (I can do it) to remove WrapTx, and also add GetGas on tx.Tx to make simulation work.

Reply:
++